Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+Conversions.h

```diff
@@ -54,7 +59,12 @@ template<auto R, typename V> struct ConversionDataSpecializer<CSS::LengthRaw<R,
                 ? state.cssToLengthConversionData().copyWithAdjustedZoom(1.0f)
                 : state.cssToLengthConversionData();
         } else if constexpr (R.zoomOptions == CSS::RangeZoomOptions::Unzoomed) {
-            return state.cssToLengthConversionData().copyWithAdjustedZoom(1.0f);
+            if (shouldUseEvaluationTimeZoom(state))
+                return state.cssToLengthConversionData().copyWithAdjustedZoom(1.0f);
+
+            return state.useSVGZoomRulesForLength()
+                ? state.cssToLengthConversionData().copyWithAdjustedZoom(1.0f)
+                : state.cssToLengthConversionData();
```

Comment:
Can you just take the previous branch of the `if` (and make it non-constexpr) in the `!shouldUseEvaluationTimeZoom()` case and avoid duplicating the code?